workers/unity/Assets/Fps/Scripts/Editor/GameLogic/MapQualitySettingsInspector.cs

```diff
@@ -0,0 +1,169 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+[CustomEditor(typeof(MapQualitySettings))]
+[CanEditMultipleObjects]
+public class MapQualitySettingsInspector : Editor
+{
+    private GUIStyle activeQualityStyle;
+
+    private int activeQualityLevel;
+
+    public override void OnInspectorGUI()
+    {
+        activeQualityLevel = QualitySettings.GetQualityLevel();
+
+        var previewProp = serializedObject.FindProperty("Preview");
+
+        RefreshQualityProperties();
+
+        DrawPropertiesExcluding(serializedObject,
+            serializedObject.FindProperty("Settings").name,
+            previewProp.name);
+        GUILayout.Space(10);
+
+        GUILayout.Label("Tile draw distances", EditorStyles.boldLabel);
+        DrawTileQualityProperties();
+
+        GUILayout.Space(10);
+
+        EditorGUILayout.PropertyField(previewProp);
+
+        if (previewProp.boolValue)
+        {
+            DrawQualityPreviewButtons();
+        }
+
+        serializedObject.ApplyModifiedProperties();
+    }
+
+    private void RefreshQualityProperties()
+    {
+        foreach (var t in targets)
+        {
+            var tileSettings = t as MapQualitySettings;
+            if (QualitiesPropertyMatchesProject(tileSettings))
+            {
+                continue;
+            }
+
+            var newQualities = new List<MapQualityLevelData>();
+            foreach (var qualityName in QualitySettings.names)
+            {
+                var newTileQualitySetting = new MapQualityLevelData();
+                newTileQualitySetting.QualityName = qualityName;
+                newTileQualitySetting.CheckoutDistance = MapQualitySettings.DefaultCheckoutDistance;
+                foreach (var setting in tileSettings.Settings)
+                {
+                    if (setting.QualityName != qualityName)
+                    {
+                        continue;
+                    }
+
+                    newTileQualitySetting.CheckoutDistance = setting.CheckoutDistance;
+                    break;
+                }
+
+                newQualities.Add(newTileQualitySetting);
+            }
+
+            tileSettings.Settings = newQualities;
+        }
+    }
+
+    private void DrawTileQualityProperties()
+    {
+        var settings = target as MapQualitySettings;
```

Comment:
Given that this can't (shouldn't?) fail - use a direct cast: `(MapQualitySettings) target`

Reply:
A good resource on the [semantics of casting](https://stackoverflow.com/questions/132445/direct-casting-vs-as-operator)

Reply:
Similar comment in the function below

Reply:
Oh, thanks. I had previously thought they were the same thing and used `as` because it looks nicer : P